### DISCUSSION TITLE
Update build doc for CWTools?

### DISCUSSION THREAD

**Discussion by CamilleHbp** [bcssov/IronyModManager#335] [Q&A] [answered]
I'm sorry, I don't really know how to contact you apart from making an issue here.

I had the idea of trying to solve mod conflict in CK3 and then I saw your project. I don't have much experience with the C# ecosystem and was trying to build your project.

It seems the Nuget CWTools version you have is private. I can't get it. There is a [CWTools.CLI](https://www.nuget.org/packages/CWTools.CLI/0.0.16-alpha1) by Thomas Boby but it's not the one that is linked in the project. Your specific version, `CWTools.Irony-Private`, is required to be *alpha8* when the one in the `README.MD` is *alpha4*.

Would you be willing to update the doc so that we can build and maybe contribute? Is there a reason you haven't used the official `CWTools` since it has been updated now? Would you be open to share a discord or something so that people developping on this project could have a place to exchange? Then I could get the info and maybe update the `README.MD` with the point of view of a C# n00b.

That's a lot of questions, thanks a lot for you time on this project!

**Comment by bcssov** [comment, marked as answer]
Better to put this in discussion section.

> I had the idea of trying to solve mod conflict in CK3 and then I saw your project. I don't have much experience with the C# ecosystem and was trying to build your project.

Irony does not support CK3 though. I mean it does for mod management, but it doesn't know how to parse CK3 structure to detect duplicates. It should be capable to parse CK3 script though.

> It seems the Nuget CWTools version you have is private. I can't get it. There is a [CWTools.CLI]
(https://www.nuget.org/packages/CWTools.CLI/0.0.16-alpha1) by Thomas Boby but it's not the one that is linked in the project. Your specific version, CWTools.Irony-Private, is required to be alpha8 when the one in the README.MD is alpha4.

I'm lazy and don't get asked often about how to build this project so I kinda neglect this piece of info. Here you can find the latest nuget:
https://github.com/bcssov/IronyModManager/discussions/327

> Would you be willing to update the doc so that we can build and maybe contribute? 

See above

> Is there a reason you haven't used the official CWTools since it has been updated now?

The reason can be answered in 2 screenshots:
![2022-02-14_15-35-02](https://user-images.githubusercontent.com/5864390/153884107-834f530b-79e3-407d-9688-fba161d882b2.png)
![2022-02-14_15-35-27](https://user-images.githubusercontent.com/5864390/153884117-a79065a5-45ae-4d50-b6dd-5b466b0dfbd8.png)
As you can see it is out of date and the author is incredibly busy and I don't want to bother him so I just build a private version which is the same as the official one. 

It's built from this branch:
https://github.com/bcssov/cwtools/tree/irony

> Would you be open to share a discord or something so that people developping on this project could have a place to exchange? 

I've never had the need for an official discord, I lack the spare time to maintain it (as in moderate it). But I do have a channel over at Stellaris Modding Den which can fulfill this purpose I guess: https://github.com/bcssov/IronyModManager/discussions/230

I think this covers all your questions.

**Reply by CamilleHbp** [reply]
I think I misunderstandd the duality of CWTools package and CWTools.CLI. I'm guessing the CLI has been updated but the package for the CWTools library itself has not?

Thank you very much for your help and all this. I will try to delve into your code and understand it. I would like to try my hand at improving it for CK3 instead of making yet another launcher substitute. Thanks!

**Comment by bcssov** [comment]
> I think I misunderstandd the duality of CWTools package and CWTools.CLI. I'm guessing the CLI has been updated but the package for the CWTools library itself has not?

Yup. I don't use the CLI project I directly reference CWTools for it's parsing capabilities, well nowadays just for validation. Irony has its own parser (a simplified dumb parser).

This is provided by CWTools:
![Image2](https://user-images.githubusercontent.com/5864390/153891842-55997a51-4467-4a76-b30d-a9d7035c28d4.jpg)

This is by Irony's own parser:
![Image4](https://user-images.githubusercontent.com/5864390/153891912-7ac2f77e-1964-43d9-9285-6a004942059e.jpg)
![Image6](https://user-images.githubusercontent.com/5864390/153891957-34322a92-23b9-4a0e-8e3b-5f0aa311faa2.jpg)


> Thank you very much for your help and all this. I will try to delve into your code and understand it. I would like to try my hand at improving it for CK3 instead of making yet another launcher substitute. Thanks!

I don't see a problem with another launcher though (more choice is always nice for people). I do plan on one day adding CK3 analysis support, it's on the roadmap actually. Just recently I've added analysis support for HOI4. Until then you can only manage mods with it in CK3.

Though Stellaris is the only one that supports full feature set for the Conflict Solver (ability to resolve conflicts). Since it has vast community resources to help me with this. The others lack this piece of information: https://stellaris.paradoxwikis.com/Modding#Overwriting_Specific_Elements

I was given some information about how CK3 behaves but as of this moment I haven't been able to come around on testing this on my friends computer.

**Reply by CamilleHbp** [reply]
Thank you for the extensive answer! Looks really nice. I'll try to take a look in more depth in the following days. I will try to juggle this and learning CK3 modding at the same time, hopefully I'll manage both in the long run!
